.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,32 @@
+# Only run on merges to trunk
+name: Build and Push Image
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build_and_push:
+    runs-on: ubuntu-latest
+  
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set vars
+        run: |
+          DOCKER_TAG=$(date '+%Y%m%d.%H%M%S')
+          echo "DOCKER_TAG=$DOCKER_TAG" >> $GITHUB_ENV
+      
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          context: ./resource-synchronizer
+          pull: true
+          push: true
+          tags: openstax/content-synchronizer:${{ env.DOCKER_TAG }}
```

Comment:
🤔 - How would the sync pipelines set with [this configuration](https://github.com/openstax/content-synchronizer/blob/987db036c43947df3d8e29b6c6d7cebde40770b5/pipeline-sync.yml#L45) consume the updated image?

Reply:
My assumption was that it would be like so:

```
    - task: sync content
        config:
          platform: linux
          image_resource:
            type: registry-image
            source:
              <<: *docker-credentials
              repository: openstax/content-synchronizer
              tag: 20210818.213827
```

Reply:
We'd obviously have to merge the changes, allow the images to push, and then submit another PR with the updated tags.

Reply:
So all of the pipelines would need to be re-set (code PR -> merge -> pipeline PR -> merge -> set N pipelines)? I'm only clarifying this as it's a different workflow than what we have currently, which is fine, just wanted to be sure it was intentional. Also, I believe the second merge would create a new tag (though with the same container contents), which means the latest pipeline would never use the latest docker tag I guess 🤷‍♂️ .

Reply:
Yea, I figured to get things in a state moving forward we'd remove these in-between tags. I wasn't sure if we wanted existing pipelines to automatically pick up changes by overwritting main. I was mainly trying to come up with a way to pin these images. Is there a different workflow we should use? My main goal was to automate this push and to pin the version.